pythonx/jedi_vim.py

```diff
@@ -765,9 +765,12 @@ def show_call_signatures(signatures=()):
     if int(vim_eval("has('conceal') && g:jedi#show_call_signatures")) == 0:
         return
 
+    # We need to clear the signatures before we calculate them again. The
+    # reason for this is that call signatures are unfortunately written to the
+    # buffer.
+    clear_call_signatures()
     if signatures == ():
         signatures = get_script().call_signatures()
-    clear_call_signatures()
```

Comment:
This should be needed only inside the `if` probably (strictly speaking)?
(it's ok to keep it like that for now though)

Reply:
True. We could move it there.

This was just my poor man's fix to avoid any regressions. If you want to further improve this, feel free. I didn't put a lot of thought into it. It's a simple move that should not change anything (except fix the original issue).